infra/gravity/cluster_status.go

```diff
@@ -2,45 +2,108 @@ package gravity
 
 import (
 	"context"
-	"time"
 
+	"github.com/cenkalti/backoff"
+
+	"github.com/gravitational/robotest/lib/constants"
 	sshutils "github.com/gravitational/robotest/lib/ssh"
 	"github.com/gravitational/robotest/lib/utils"
 	"github.com/gravitational/robotest/lib/wait"
-
 	"github.com/gravitational/trace"
 )
 
-// Status walks around all nodes and checks whether they all feel OK
-func (c *TestContext) Status(nodes []Gravity) error {
-	c.Logger().WithField("nodes", Nodes(nodes)).Info("Check status on nodes.")
-	ctx, cancel := context.WithTimeout(c.ctx, c.timeouts.Status)
-	defer cancel()
+// returns nil if the Gravity Status is the expected status or an error otherwise
+type statusValidator func(s GravityStatus) error
 
-	retry := wait.Retryer{
-		Attempts: 100,
-		Delay:    time.Second * 20,
+// a negation / reimplementation of https://github.com/gravitational/gravity/blob/7.0.0/lib/status/status.go#L180-L185
+// I'd prefer isDegraded and negation later, but I didn't find an elegant way to
+// have consumer friendly error messages when the information channel is squashed to != nil
+// -- 2020-05 walt
+func isNotDegraded(s GravityStatus) error {
```

Comment:
Nit: Speaking about semantics, I think that functions named like "isXXX" are rather expected to return a boolean value. Otherwise it might be a bit confusing i.e. does this return an error if it's degraded or not?

I would consider renaming this to "checkNotDegraded" or "assertNotDegraded".